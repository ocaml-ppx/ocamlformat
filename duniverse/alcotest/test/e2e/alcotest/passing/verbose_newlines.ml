(** Reproduction for https://github.com/mirage/alcotest/issues/225, testing the
    interaction between `--verbose` and newlines in the test stdout. *)

let () =
  Alcotest.run ~verbose:true __FILE__
    [
      ( "alpha",
        [
          Alcotest.test_case "0 newlines" `Quick (fun () ->
              Format.printf "Print inside alpha");
        ] );
      ( "beta",
        (* Progressive result reporting is broken for this test, since the
           carriage return happens on the wrong line. *)
        [
          Alcotest.test_case "1 newline" `Quick (fun () ->
              Format.printf "Print inside beta\n");
        ] );
      ( "gamma",
        [
          (* Reporting is also broken here. Even worse, some of the test std.out
             is clipped by the eventual result of the test. *)
          Alcotest.test_case "1 newline + long line" `Quick (fun () ->
              Format.printf
                "Print inside gamma\n\
                 Lorem ipsum dolor sit amet, consectetur adipiscing elit, \
                 nullam malesuada dictum tortor in venenatis.");
        ] );
      ( "delta",
        [
          Alcotest.test_case "1 newline + long check" `Quick (fun () ->
              Format.printf "Print inside delta\n";
              Alcotest.(check unit)
                "Lorem ipsum dolor sit amet, consectetur adipiscing elit, \
                 nullam malesuada dictum tortor in venenatis."
                () ());
        ] );
    ]
