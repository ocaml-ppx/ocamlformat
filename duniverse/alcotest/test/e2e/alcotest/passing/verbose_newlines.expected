Testing `test/e2e/alcotest/passing/verbose_newlines.ml'.
This run has ID `<uuid>'.

Print inside alpha  [OK]          alpha          0   0 newlines.
Print inside beta
  [OK]          beta           0   1 newline.
Print inside gamma
Lorem ipsum dolor sit amet, consectetur adipiscing elit, nullam malesuada dictum tortor in venenatis.  [OK]          gamma          0   1 newline + long line.
Print inside delta
ASSERT Lorem ipsum dolor sit amet, consectetur adipiscing elit, nullam malesuada dictum tortor in venenatis.
  [OK]          delta          0   1 newline + long check.

Test Successful in <test-duration>s. 4 tests run.
